dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java

```diff
@@ -636,6 +639,18 @@ public String getKey()
             .addIgnoreNull( "apiVersion", apiVersion ).build();
     }
 
+    private String getDimensionalItemKeywords( final DimensionItemKeywords keywords )
+    {
+        final StringBuilder keys = new StringBuilder();
+
+        if ( keywords != null && !keywords.isEmpty() )
+        {
+            keywords.getKeywords().forEach( k -> keys.append( "key:" ).append( "[" + k.getKey() + "]" ) );
+        }
```

Comment:
Could be simplified with something like:

 ```
 return keywords.getKeywords().stream()
            .map( Keyword::getKey )
            .collect( Collectors.joining( ":" ) );
```

Reply:
Done.